src/ccache/argprocessing.cpp

```diff
@@ -1090,7 +1090,10 @@ process_option_arg(const Context& ctx,
     return Statistic::none;
   }
 
-  if (compopt_takes_path(arg)) {
+  // -Yc allows concatenated filepath argument only (without space) and this
+  // filepath is optional, this if() block can't be invoked if the -Yc argument
+  // was passed without a filepath.
+  if (compopt_takes_path(arg) && arg != "-Yc") {
```

Comment:
Could this be

```suggestion
  if (compopt_takes_arg(arg) && compopt_takes_path(arg)) {
```

instead?

Reply:
This is much better solution thx, I didn't like that `arg != "-Yc"`. All other msvc options should be good.